File: 1-two-sum/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        numToIdx = {}

        for i, n in enumerate(nums):
            if target-n in numToIdx:
                return [i, numToIdx[target-n]]
            numToIdx[n] = i

File: 1-two-sum/1. Two Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        
        var indices: [Int: Int] = [:] //nums[i] : i
        
      
        for i in 0..<nums.count {
            if let j = indices[target - nums[i]] {
                return [j, i]
            }
            indices[nums[i]] = i
        }



        return [-1]
    }
}